sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py

```diff
@@ -236,10 +239,14 @@ async def _inner_anext(self):
                 raise
 
     async def __anext__(self):
-        message = await self._inner_anext()
-        links = get_receive_links(message)
-        with receive_trace_context_manager(self, links=links):
-            return message
+        try:
+            self._receive_context.set()
```

Comment:
Is this not using an `asyncio.Event`?

Reply:
nope, I'm reusing the threading.Event because the only feature we used is `set` and `clear` which threading.Event seem to satisfy our goal here.

but since you called it out, I think we could change it to asyncio.Event for keep async consistency.

Reply:
code updated to asyncio.Event